pkg/git/libgit2/transport.go

```diff
@@ -126,20 +129,24 @@ func (s *PublicKeyAuth) Method(secret corev1.Secret) (*git.Auth, error) {
 		user = git.DefaultPublicKeyAuthUser
 	}
 
-	credCallback := func(url string, username_from_url string, allowed_types git2go.CredType) (*git2go.Cred, error) {
+	credCallback := func(url string, usernameFromURL string, allowedTypes git2go.CredType) (*git2go.Cred, error) {
 		cred, err := git2go.NewCredSshKeyFromMemory(user, "", string(identity), "")
 		if err != nil {
 			return nil, err
 		}
 		return cred, nil
 	}
 	certCallback := func(cert *git2go.Certificate, valid bool, hostname string) git2go.ErrorCode {
+		host, _, err := net.SplitHostPort(s.host)
+		if err != nil || host != hostname {
```

Comment:
`SplitHostPort` will return an error if there's no port in `s.host`, so this will fail in the case of `github.com`. I think we still want the sanity check against the passed-in hostname, so the way to do this is:

```go
host, _, err := net.SplitHostPort(s.host)
if err != nil {
  host = s.host
}
if host != hostname {
  return git2go.ErrUser
}

host = knownhosts.Normalize(s.host) // <-- NB start from s.host again
for _, k := range kk {
	if k.matches(host, cert.Hostkey.HashSHA1[:]) {
// ...etc
```